aips/aip-53.md

```diff
@@ -0,0 +1,54 @@
+---
+aip: 53
+title: Make Fee Payer Address Optional in Transaction Authenticator
+author: davidiw
+discussions-to (*optional): <a url pointing to the official discussion thread>
+Status: Draft
+last-call-end-date (*optional): <mm/dd/yyyy the last date to leave feedbacks and reviews>
+type: Framework
+created: 10/08/2023
+updated (*optional): <mm/dd/yyyy>
+requires (*optional): 39
+---
+
+# AIP-53 - Make Fee Payer Address Optional in Transactions Authenticator
+
+## Summary
+
+When submitting sponsored transactions to the blockchain, a client may not know the actual fee payer and likely does not care. The existing mechanism to generate these transactions requires the client to know prior to sending the transaction to the blockchain the payer. This impedes developer experience for limited value. This AIP proposes making the knowledge of the ultimate payer optional.
+
+### Goals
+
+This AIP makes it easier for a client to submit a sponsored transaction and be ignorant of who ultimately pays for it. As result, this will substantially improve the developer experience on Aptos.
+
+## Alternative solutions
+
+While there are a few discussions on refactoring the transaction layout to eliminate legacy quirks introduced by the long history of the Aptos code base, those proposals have yet to be published and will regretablly take substantial effort to gain adoption. Furthermore, these updates would require changes across many pieces within the Aptos software stack including SDKs, Indexers, as well as transaction and Move code resulting in several months of dedicated engineering ignoring the social challenges of pushing such an update.
+
+## Specification
+
+Currently, when signing a transaction, the fee payer's address is included within the data to be signed. As a result, all parties must first know the fee payer's address prior to being able to submit a transaction.
+
+This AIP dictates that the fee payer's address may either be set to the actual address or to `0x0`, a non-allocated address within Move. As a result, during verification, each signature must be compared against the raw transaction computed with the actual fee payer's address and that of `0x0`.
+
+## Reference Implementation
+
+https://github.com/aptos-labs/aptos-core/pull/10443
+
+## Testing
+
+Verified via end-to-end tests.
+
+## Risks and Drawbacks
+
+There may be a slight performance degradation for sponsored transactions as now many formats need to be checked. These tend to be cheap and at worst case double the cost of verification. As most SDKs will likely conform to the simpler method of using `0x0`, this will inevitably become a negligible cost. Once we update to transaction v2, this check can be eliminated.
```

Comment:
what is transaction v2?

Reply:
And do we have number comparison for the performance degradation? how much more costly we're talking about here?